src/test/java/io/jenkins/plugins/grading/AutoGraderTest.java

```diff
@@ -37,4 +40,42 @@ void shouldThrowNullPointerExceptionOnValidButNotCorrectConfiguration() {
                         TaskListener.NULL));
     }
 
+    @Test
+    void shouldLogScoreFromRecordedTestResults() {
+        String testConfiguration = "{ 'tests' : "
+                + "{ "
+                + "'passedImpact' : 1, "
+                + "'failureImpact' : 1, "
+                + "'skippedImpact' : 1, "
+                + "'maxScore' : 1 "
+                + "} "
+                + "}";
+        String displayName = "testName";
+        AutoGrader autoGrader = new AutoGrader(testConfiguration);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        TestResultAction action = mock(TestResultAction.class);
+        when(action.getFailCount()).thenReturn(1);
+        when(action.getSkipCount()).thenReturn(1);
+        when(action.getTotalCount()).thenReturn(5);
+        when(action.getDisplayName()).thenReturn(displayName);
+
+        AggregatedScore score = mock(AggregatedScore.class);
+        when(score.addTestsTotal(any(), any())).thenReturn(1);
+
+        TestScore testScore = mock(TestScore.class);
+        when(testScore.getTotalImpact()).thenReturn(1);
+
+        Run run = mock(Run.class);
+        when(run.getAction(any())).thenReturn(action);
+
+        autoGrader.perform(run, new FilePath((VirtualChannel) null, "/"), mock(Launcher.class),
+                TaskListener.NULL);
+
+
+        Assertions.assertThat(outputStream.toString())
+                .isEqualTo(String.format("Grading test results [%s]\n"
+                        + "-> Score [%d] - from recorded test results: [%d], [%d], [%d], [%d]\n"
+                        + "Total score for test results: [%s]",displayName, 1, 1, 1, 1, 1, 1));
```

Comment:
War fast richtig. Da fehlte noch der Dateipfad wo die Datei abgelegt wird. Ich habe zusätzlich auch noch den PrintStream rausgezogen, so dass Sie mit Strings prüfen können.

```suggestion
        
        Path tempFile = Files.createTempFile("score", "xml");
        when(run.getRootDir()).thenReturn(tempFile.toFile());

        ByteArrayOutputStream out = new ByteArrayOutputStream();
        PrintStream printStream = new PrintStream(out, true, StandardCharsets.UTF_8.name());

        TaskListener listener = mock(TaskListener.class);
        when(listener.getLogger()).thenReturn(printStream);

        autoGrader.perform(run, new FilePath((VirtualChannel) null, "/"), mock(Launcher.class), listener);

        assertThat(out.toString()).contains("... TODO ...");
```